@xen-orchestra/backups/writers/_abstractExportWriter.js

```diff
@@ -0,0 +1,15 @@
+const { asyncMap } = require('@xen-orchestra/async-map')
+const { createLogger } = require('@xen-orchestra/log')
+
+const { BACKUP_DIR } = require('../_getVmBackupDir')
+
+const { debug } = createLogger('xo:backups:abstractExportWriter')
+
+exports.abstractExportWriter = {
+  async cleanCorruptedBackups() {
+    const adapter = this._adapter
+    await asyncMap(await adapter.handler.list(BACKUP_DIR, { prependDir: true }), vmDir =>
+      adapter.cleanVm(vmDir, { remove: true, merge: true, onLog: debug })
+    )
+  },
+}
```

Comment:
Maybe it would be interesting to implement this as a [mixin class](https://exploringjs.com/impatient-js/ch_proto-chains-classes.html#mixin-classes-advanced)?

Reply:
Maybe `AbstractDeltaWriter` and `AbstractFullWriter` that should be implemented as a mixins classes according to the above comment.